simexpal/base.py

```diff
@@ -1013,22 +1018,32 @@ def source_dir(self):
 
 	def is_checked_out(self):
 		if self.revision.is_dev_build:
-			return os.access(os.path.join(self.source_dir, 'checkedout.simexpal'), os.F_OK)
+			with open(self._cfg.basedir + '/simex.cache', 'r') as myfile:
+				return json.load(myfile)["checkout_done"]
+			#return os.access(os.path.join(self.source_dir, 'checkedout.simexpal'), os.F_OK)
 		return os.access(os.path.join(self.clone_dir, 'checkedout.simexpal'), os.F_OK)
 
 	def is_regenerated(self):
 		if self.revision.is_dev_build:
-			return os.access(os.path.join(self.source_dir, 'regenerated.simexpal'), os.F_OK)
+			with open(self._cfg.basedir + '/simex.cache', 'r') as myfile:
+				return json.load(myfile)["regeneration_done"]
+			#return os.access(os.path.join(self.source_dir, 'regenerated.simexpal'), os.F_OK)
 		return os.access(os.path.join(self.clone_dir, 'regenerated.simexpal'), os.F_OK)
 
 	def is_configured(self):
-		return os.access(os.path.join(self.compile_dir, 'configured.simexpal'), os.F_OK)
+		with open(self._cfg.basedir + '/simex.cache', 'r') as myfile:
+			return json.load(myfile)["configuration_done"]
```

Comment:
I see that we have to repeat these JSON keys several times. I recommend having constants defining those keys and using them as variables. This way we can't trip over misspellings.